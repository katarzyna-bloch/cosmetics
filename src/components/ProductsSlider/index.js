import PropTypes from 'prop-types'
import styled from 'styled-components'

import Carousel from '../Carousel'
import { sizes } from '../../utils/media'
import ProductListItem, { Image } from '../ProductListItem'

const ListItem = styled(ProductListItem)`
  padding: 0;

  ${Image} {
    height: 260px;
  }
`

export const ProductsSlider = ({ products }) => {
  const items = products.map(product => (
    <ListItem product={product} key={product.id} />
  ))

  return (
    <Carousel
      items={items}
      settings={{
        slidesToShow: 3,
        slidesToScroll: 2,
        responsive: [
          {
            breakpoint: sizes.tablet,
            settings: {
              slidesToShow: 2,
              slidesToScroll: 2,
              infinite: true,
            }
          },
          {
            breakpoint: sizes.phone,
            settings: {
              slidesToShow: 1,
              slidesToScroll: 1
            }
          } 
        ],
        gap: 30,
      }}
    />
  )
}

ProductsSlider.propTypes = {
  products: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    price: PropTypes.string.isRequired,
    oldPrice: PropTypes.string,
    hasDiscount: PropTypes.bool,
  })).isRequired,
}

export default ProductsSlider
