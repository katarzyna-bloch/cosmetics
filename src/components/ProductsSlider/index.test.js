import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ProductsSlider from '.'

jest.mock('../../components/Carousel', () => {
  const Carousel = () => null
  return Carousel
})

const props = {
  products: [
    {
      brand: "Eigshow",
      categoryId: "3",
      description: "test opisu",
      id: "23",
      img: "pędzle.jpg",
      name: "Zestaw pędzli",
      price: "73 zł",
      rate: 8,
      volume: "120 ml",
    }
  ],
}

it('should render correctly', () => {
  const wrapped = shallow(<ProductsSlider {...props} />)
  expect(toJson(wrapped)).toMatchSnapshot()
})
