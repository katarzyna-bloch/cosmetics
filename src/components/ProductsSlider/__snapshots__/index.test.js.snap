// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Carousel
  items={
    Array [
      <Styled(ProductListItem)
        product={
          Object {
            "brand": "Eigshow",
            "categoryId": "3",
            "description": "test opisu",
            "id": "23",
            "img": "pędzle.jpg",
            "name": "Zestaw pędzli",
            "price": "73 zł",
            "rate": 8,
            "volume": "120 ml",
          }
        }
      />,
    ]
  }
  settings={
    Object {
      "gap": 30,
      "responsive": Array [
        Object {
          "breakpoint": 768,
          "settings": Object {
            "infinite": true,
            "slidesToScroll": 2,
            "slidesToShow": 2,
          },
        },
        Object {
          "breakpoint": 414,
          "settings": Object {
            "slidesToScroll": 1,
            "slidesToShow": 1,
          },
        },
      ],
      "slidesToScroll": 2,
      "slidesToShow": 3,
    }
  }
/>
`;
